File: 성효경/1주차/Boj_10866.java
import java.io.*;
import java.util.*;

public class Boj_10866 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        Deque<Integer> d = new ArrayDeque<>();
        int n = Integer.parseInt(br.readLine());

        for (int i=0; i<n; i++) {
            String s = br.readLine();
            String[] order = s.split(" ");

            switch (order[0]) {

                case "push_front":
                    int x = Integer.parseInt(order[1]);
                    d.offerFirst(x);
                    break;

                case "push_back":
                    x = Integer.parseInt(order[1]);
                    d.offerLast(x);
                    break;

                case "pop_front":
                    if (d.isEmpty()) {
                        System.out.println(-1);
                    } else {
                        System.out.println(d.pollFirst());
                    }
                    break;

                case "pop_back":
                    if (d.isEmpty()) {
                        System.out.println(-1);
                    } else {
                        System.out.println(d.pollLast());
                    }
                    break;

                case "size":
                    System.out.println(d.size());
                    break;

                case "empty":
                    if (d.isEmpty()) {
                        System.out.println(1);
                    } else {
                        System.out.println(0);
                    }
                    break;

                case "front":
                    if (d.isEmpty()) {
                        System.out.println(-1);
                    } else {
                        System.out.println(d.getFirst());
                    }
                    break;

                case "back":
                    if (d.isEmpty()) {
                        System.out.println(-1);
                    } else {
                        System.out.println(d.getLast());
                    }
                    break;
            }
        }
    }
}

File: 2주차/Boj_10866.cs
using System;
using System.Collections.Generic;

class Program
{
    static void Boj_10866(string[] args)
    {
        int N = int.Parse(Console.ReadLine());
        Deque<int> deque = new Deque<int>();

        for (int i = 0; i < N; i++)
        {
            string[] command = Console.ReadLine().Split();

            if (command[0] == "push_front")
            {
                int X = int.Parse(command[1]);
                deque.PushFront(X);
            }
            else if (command[0] == "push_back")
            {
                int Y = int.Parse(command[1]);
                deque.PushBack(Y);
            }
            else if (command[0] == "pop_front")
            {
                if (!deque.IsEmpty())
                {
                    Console.WriteLine(deque.PopFront());
                }
                else
                {
                    Console.WriteLine(-1);
                }
            }
            else if (command[0] == "pop_back")
            {
                if (!deque.IsEmpty())
                {
                    Console.WriteLine(deque.PopBack());
                }
                else
                {
                    Console.WriteLine(-1);
                }
            }
            else if (command[0] == "size")
            {
                Console.WriteLine(deque.Size());
            }
            else if (command[0] == "empty")
            {
                Console.WriteLine(deque.IsEmpty() ? 1 : 0);
            }
            else if (command[0] == "front")
            {
                if (!deque.IsEmpty())
                {
                    Console.WriteLine(deque.Front());
                }
                else
                {
                    Console.WriteLine(-1);
                }
            }
            else if (command[0] == "back")
            {
                if (!deque.IsEmpty())
                {
                    Console.WriteLine(deque.Back());
                }
                else
                {
                    Console.WriteLine(-1);
                }
            }
        }
    }
}

class Deque<T>
{
    private LinkedList<T> list = new LinkedList<T>();

    public void PushFront(T item)
    {
        list.AddFirst(item);
    }

    public void PushBack(T item)
    {
        list.AddLast(item);
    }

    public T PopFront()
    {
        T item = list.First.Value;
        list.RemoveFirst();
        return item;
    }

    public T PopBack()
    {
        T item = list.Last.Value;
        list.RemoveLast();
        return item;
    }

    public int Size()
    {
        return list.Count;
    }

    public bool IsEmpty()
    {
        return list.Count == 0;
    }

    public T Front()
    {
        return list.First.Value;
    }

    public T Back()
    {
        return list.Last.Value;
    }
}
